src/hal/src/image.rs

```diff
@@ -529,67 +535,81 @@ impl From<RenderDesc> for DepthStencilDesc {
     }
 }
 
+/// Specifies options for how memory for an image is arranged.
+/// These are hints to the GPU driver and may or may not have actual
+/// performance effects, but describe constraints on how the data
+/// may be used that a program *must* obey.  They do not specify
+/// how channel values or such are laid out in memory; the actual
+/// image data is considered opaque.
 ///
+/// Details may be found in [the Vulkan spec](https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#resources-image-layouts)
 #[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum ImageLayout {
-    ///
+    /// General purpose, no restrictions on usage.
     General,
-    ///
+    /// Must only be used as a color attachment in a framebuffer.
     ColorAttachmentOptimal,
-    ///
+    /// Must only be used as a depth attachment in a framebuffer.
     DepthStencilAttachmentOptimal,
-    ///
+    /// Must only be used as a depth attachment in a framebuffer,
+    /// or as a read-only depth or stencil buffer in a shader.
     DepthStencilReadOnlyOptimal,
-    ///
+    /// Must only be used as a read-only image in a shader.
     ShaderReadOnlyOptimal,
-    ///
+    /// Must only be used as the source for a transfer command.
     TransferSrcOptimal,
-    ///
+    /// Must only be used as the destination for a transfer command.
     TransferDstOptimal,
-    ///
+    /// No layout, does not support device access.  Only valid as a
+    /// source layout when transforming data to a specific destination
+    /// layout or initializing data.  Does NOT guarentee that the contents 
+    /// of the source buffer are preserved.
     Undefined, //TODO: consider Option<> instead?
-    ///
+    /// Like `Undefined`, but does guarentee that the contents of the source
+    /// buffer are preserved.
     Preinitialized,
-    ///
+    /// The layout that an image must be in to be presented to the display.
     Present,
 }
 
 bitflags!(
-    ///
+    /// Bitflags to describe how memory in an image or buffer can be accessed.
     #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
     pub struct Access: u16 {
         /// Read state but can only be combined with `COLOR_ATTACHMENT_WRITE`.
         const COLOR_ATTACHMENT_READ = 0x1;
         /// Write-only state but can be combined with `COLOR_ATTACHMENT_READ`.
         const COLOR_ATTACHMENT_WRITE = 0x2;
-        ///
+        /// Read access to the buffer in a copy operation.
         const TRANSFER_READ = 0x4;
-        /// Write-only state of copy commands.
+        /// Write access to the buffer in a copy operation.
         const TRANSFER_WRITE = 0x8;
-        /// Read-only state for SRV access, or combine with `SHADER_WRITE` to have r/w access to UAV.
+        /// Read-only state for shader access, or combine with `SHADER_WRITE` to have r/w access to UAV.
```

Comment:
hmm I actually chose the `SRV` term to differ it from `UAV` read access, which we currently can't handle with `SHADER_READ` only.